cmd/dockerd/daemon.go

```diff
@@ -238,6 +240,17 @@ func (cli *DaemonCli) start(opts *daemonOptions) (err error) {
 
 	setOTLPProtoDefault()
 	otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{}))
+
+	// Taken from https://github.com/moby/buildkit/blob/v0.12.4/util/tracing/detect/detect.go#L100-L107
+	res, err := resource.Detect(context.Background(), serviceNameDetector{})
```

Comment:
Do we still need this, or is "resource.Default()" enoug?

Reply:
Thanks for looking! Let me push a commit to change that.

I'll keep changes as separate commit, but will squash things before merging (just keeping my own sanity a bit 😂)